Backend/ReleaseProcess.md

```diff
@@ -0,0 +1,46 @@
+Release process 📦
+==================================
+
+Our backend has a micro-service architecture. 
+To help improve the release process we are using [Semantic Release](https://semantic-release.gitbook.io/semantic-release/) in our commits.
+
+To automate finding the next version, we use [go-semrel-gitlab](https://juhani.gitlab.io/go-semrel-gitlab/) in our pipelines.
+
+
+## Merge requests
+
+Every merge request should run a pipeline to build, test and lint those changes, and only when everything is ✅ the merge can be done. 
+
+The stages for the merge requests are:
+
+![Merge request](./media/merge_request.png)
+
+- `BuildMR` - Build or assemble the project
+- `LintMR` - Run lint checks
+- `TestsMR` - Run tests 
+
+
+## After merge
+
+After the merge is merged (🤯), a new set of jobs should run, to effectively release the new build.
+
+The stages are: 
+
+![Merge request](./media/merged.png)
+
+- `Version` - Find the new version for the build. As we do not deploy automatically to **production**, we use a build number for the releases. Usually we use [GitHub flow](https://guides.github.com/introduction/flow/), so the build number could be the number of commits in the `master` branch.
+
+- `Build` - Build the project and produce the binary (.jar, .exe 🤢, if applied ), should the defined as an artifact.
```

Comment:
i would remove the binaries types, or at least the .exe it's not used in our case.

but the english is weird 🤔 ... should the defined as an artifact? 🤔

Reply:
The types were just as examples, but I'll remove them.

About the artifact stuff, I was trying to say that it should be defined in the job as an artifact, to be used in the next job, to produce the Docker.

Do you think that this is not needed? 🤔

Reply:
ok ok, i agree with the artifact, it makes sense.